src/hotspot/share/code/debugInfo.hpp

```diff
@@ -295,14 +295,14 @@ class MonitorValue: public ResourceObj {
 // DebugInfoReadStream specializes CompressedReadStream for reading
 // debugging information. Used by ScopeDesc.
 
-class DebugInfoReadStream : public CompressedReadStream {
+class DebugInfoReadStream : public CompressedSparseDataReadStream {
```

Comment:
I don't think `DebugInfoReadStream`/`DebugInfoWriteStream` need public inheritance. The relation  is more like composition.
I would have implemented them like:
```
class DebugInfoReadStream : private CompressedSparseDataReadStream {
public:
   // we are using only needed functions from CompressedSparseDataReadStream.
    using CompressedSparseDataReadStream::buffer();
    using CompressedSparseDataReadStream::read_int();
    using ...
};
```
Or
```
template <typename DataReadStream> class DebugInfoReadStream {
public:
    // define only needed functions which use a minimum number of functions from DataReadStream
};
```

I prefer the templates because we can easily switch between different implementations of `DataReadStream`/DataWriteStream` without doing this kind of modifications.

Reply:
@No templates please! :)
For me, the following change is counterproductive as well.
```
- class DebugInfoReadStream : private CompressedSparseDataReadStream {
+ class DebugInfoReadStream : private CompressedSparseDataReadStream {
+  public:
+   using CompressedSparseDataReadStream::read_int;
+   using CompressedSparseDataReadStream::read_signed_int;
+   using CompressedSparseDataReadStream::read_double;
+   using CompressedSparseDataReadStream::read_long;
+   using CompressedSparseDataReadStream::read_bool;
+   using CompressedSparseDataReadStream::CompressedSparseDataReadStream;
```